libusb/os/darwin_usb.c

```diff
@@ -866,14 +866,35 @@ static int get_device_port (io_service_t service, UInt8 *port) {
   return ret;
 }
 
+static int get_device_parent_sessionID(io_service_t service, UInt64 *parent_sessionID) {
+  kern_return_t result;
+  io_service_t parent;
+
+  /* First try to find a parent in the IOUSB plane */
+  result = IORegistryEntryGetParentEntry (service, kIOUSBPlane, &parent);
+  if (kIOReturnSuccess == result) {
+    return get_ioregistry_value_number (parent, CFSTR("sessionID"), kCFNumberSInt64Type, parent_sessionID);
+  }
```

Comment:
The first check is walking up the IOUSB plane. I don't have an El Capitan (or older) system to test the new loop on, so I figured i'd keep the old check in case it worked on those systems.